app/bundles/StageBundle/Tests/Functional/Controller/Api/StageApiControllerTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\StageBundle\Tests\Functional\Controller\Api;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\StageBundle\Entity\Stage;
+use PHPUnit\Framework\Assert;
+
+class StageApiControllerTest extends MauticMysqlTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    protected function beforeBeginTransaction(): void
+    {
+        $this->resetAutoincrement([
+            'leads',
+            'stages',
+        ]);
+    }
+
+    public function testAddStageToContact(): void
```

Comment:
I'd suggest to also create a point action and trigger in this test and assert that the point action was triggered and points accumulated. Basically follow the steps to test you wrote int he PR description.